protocol/contracts/libraries/Silo/LibTokenSilo.sol

```diff
@@ -6,193 +6,416 @@ pragma solidity =0.7.6;
-            myFunctionCall
+            callData
         );
+
         if (!success) {
             if (data.length == 0) revert();
             assembly {
                 revert(add(32, data), mload(data))
             }
         }
+
         assembly {
             bdv := mload(add(data, add(0x20, 0)))
         }
     }
 
-    function tokenWithdrawal(
+    /**
+     * @dev Locate the `amount` and `bdv` for a user's Deposit in storage.
+     * 
+     * Silo V2 Deposits are stored within each {Account} as a mapping of:
+     *  `address token => uint32 season => { uint128 amount, uint128 bdv }`
+     * 
+     * Unripe BEAN and Unripe LP are handled independently so that data
+     * stored in the legacy Silo V1 format and the new Silo V2 format can
+     * be appropriately merged. See {LibUnripeSilo} for more information.
+     *
+     * FIXME(naming): rename to `getDeposit()`?
+     */
+    function tokenDeposit(
         address account,
         address token,
-        uint32 id
-    ) internal view returns (uint256) {
+        int96 stem
+    ) internal view returns (uint256 amount, uint256 bdv) {
+        AppStorage storage s = LibAppStorage.diamondStorage();
+        bytes32 depositId = LibBytes.packAddressAndStem(
+            token,
+            stem
+        );
+        amount = s.a[account].deposits[depositId].amount;
+        bdv = s.a[account].deposits[depositId].bdv;
+    }
+    
+    /**
+     * @dev Get the number of Stalk per BDV per Season for a whitelisted token. Formerly just seeds.
+     * Note this is stored as 1e6, i.e. 1_000_000 units of this is equal to 1 old seed.
+     */
+    function stalkEarnedPerSeason(address token) internal view returns (uint256) {
         AppStorage storage s = LibAppStorage.diamondStorage();
-        return s.a[account].withdrawals[token][id];
+        return uint256(s.ss[token].stalkEarnedPerSeason);
     }
 
-    function seeds(address token) internal view returns (uint256) {
+    /**
+     * @dev Get the number of Stalk per BDV for a whitelisted token. Formerly just stalk.
+     */
+    function stalkIssuedPerBdv(address token) internal view returns (uint256) {
         AppStorage storage s = LibAppStorage.diamondStorage();
-        return uint256(s.ss[token].seeds);
+        return uint256(s.ss[token].stalkIssuedPerBdv);
     }
 
-    function stalk(address token) internal view returns (uint256) {
+    //this returns grown stalk with no decimals
+    function stemTipForToken(address token)
+        internal
+        view
+        returns (int96 _stemTipForToken)
+    {
         AppStorage storage s = LibAppStorage.diamondStorage();
-        return uint256(s.ss[token].stalk);
+        
+        _stemTipForToken = s.ss[token].milestoneStem +
+        int96(s.ss[token].stalkEarnedPerSeason).mul(
+            int96(s.season.current).sub(int96(s.ss[token].milestoneSeason))
+        ).div(1e6); //round here 
+    }
+
+    function grownStalkForDeposit(
```

Comment:
Exposing through SiloExit since the UI is probably going to want to show the grown stalk for each deposit individually, like it does now. https://github.com/BeanstalkFarms/Beanstalk/commit/e2ca5a69be899c90262150a68219e67f8700b70c

Reply:
Sorry got trigger happy on that commit button, fixed in that one and this one: https://github.com/BeanstalkFarms/Beanstalk/commit/6249f815d6f4cb20537d68fc167e97a5caba9f6a